src/scaffoldmaker/meshtypes/meshtype_3d_spinalnerve1.py

```diff
@@ -0,0 +1,705 @@
+                        x = xList[i]
+                        x = add(x, translateToSC)
+                        d1 = d1List[i]
+
+                        if nerve == 'dorsal':
+                            d2, d3, id2, id3, d12, d13, id12, id13 = \
+                                d2List[i], d3List[i], id2List[i], id3List[i], d12List[i], d13List[i], id12List[i],\
+                                id13List[i]
+                        else:
+                            d2 = set_magnitude(cross(d3, d1List[i]), magnitude(d3))
+                            id2 = mult(d2, innerProportionDefault)
+                            if i < sideNerveElementsCount:
+                                d2Next = set_magnitude(cross(d3, d1List[i + 1]), magnitude(d3))
+                                d12 = sub(d2Next, d2)
+                            else:
+                                d12 = [0.0, 0.0, 0.0]
+                            id12 = mult(d12, innerProportionDefault)
+
+                        setNodeFieldParameters(coordinates, fieldcache, x, d1, d2, d3, d12, d13)
+                        setNodeFieldParameters(innerCoordinates, fieldcache, x, d1, id2, id3, id12, id13)
+                        nodeIdentifier += 1
+
+        return annotationGroups, networkMesh
+
+    @classmethod
+    def getInteractiveFunctions(cls):
+        """
+        Edit base class list to include only valid functions.
+        """
+        interactiveFunctions = super(MeshType_1d_human_spinal_nerve_network_layout1, cls).getInteractiveFunctions()
+        for interactiveFunction in interactiveFunctions:
+            if interactiveFunction[0] == "Edit structure...":
+                interactiveFunctions.remove(interactiveFunction)
+                break
+        return interactiveFunctions
+
+
+class MeshType_3d_spinalnerve1(Scaffold_base):
+    """
+    Generates a 3-D hermite bifurcating tube network with core representing the human spinal nerve.
+    """
+
+    @classmethod
+    def getName(cls):
+        return "3D Spinal Nerve 1"
+
+    @classmethod
+    def getParameterSetNames(cls):
+        return [
+            "Default",
+            "Human 1",
+            "Human pair 1",
+            "Human whole spine 1"
+        ]
+
+    @classmethod
+    def getDefaultOptions(cls, parameterSetName="Default"):
+        options = {}
+        useParameterSetName = "Human 1" if (parameterSetName == "Default") else parameterSetName
+        options["Base parameter set"] = useParameterSetName
+        options["Spinal nerve network layout"] = ScaffoldPackage(MeshType_1d_human_spinal_nerve_network_layout1,
+                                                                 defaultParameterSetName=useParameterSetName)
+        options["Number of elements along nerve root"] = 5
+        options["Number of elements along dorsal root around DRG"] = 2
+        options["Number of elements along dorsal root ganglion"] = 4
+        options["Number of elements along ventral root"] = 8
```

Comment:
Probably too many elements, what about 5,2,4,8 -> 3,1,3,5.
Rename "along dorsal root around DRG" to "along dorsal root beside DRG", as "around" has a special meaning.